skyportal/handlers/api/photometry.py

```diff
@@ -131,7 +132,8 @@ def hash_photometry(row):
     m.update(np.int32(row["instrument_id"]))
     m.update(row["filter"].encode('utf-8'))
     m.update(np.float64(row["mjd"]))
-    m.update(np.float32(row["standardized_flux"]))
+    if row["standardized_flux"] is not None:
+        m.update(np.float32(row["standardized_flux"]))
```

Comment:
Is it safe to not update the hash when flux is none? Should we update with a dummy key in that case (to avoid potential collisions)?

Reply:
I don't see how using one dummy key would prevent collisions, tbh. I'd ask someone more knowledgeable about hashes. @stefanv @kmshin1397 what say you my friends?

Reply:
I'm not sure I understand what collisions we're trying to prevent here. If you used a random dummy key and not the same one every time, you would get different hashes for multiple instances of flux is none. Even without the dummy key though, you would only get collisions if you had two payloads with the other hash columns the same and both had `row["standardized_flux"] is None`, which would actually be a true duplicate anyways, no?

Reply:
Here is a simplified version of the collision scenario I envision:

I have 2 points, A and B, with 2 fields each, x and y. Imagine A.x, B.x, B.y are nonnull and A.y is null.

The collision scenario I imagine is:

Hash(A.x) = Hash(B.x).update(Hash(B.y))

If we modify the hash on a null value we would not get this condition:

Hash(A.x).update(Hash(DUMMY)) != Hash(B.x).update(Hash(B.y))

Where DUMMY is a constant that is the same for all y [and other fields].

Reply:
Not sure about Python hashlib and update specifically, but generally, Hash(A.x) is not the same as Hash(B.x + B.y) which I thought was what Hash(B.x).update(Hash(B.y)) is doing?

Reply:
I believe the probability of `Hash(A.x) = Hash(B.x).update(Hash(B.y))` is essentially the same as of `Hash(A.x).update(Hash(DUMMY)) == Hash(B.x).update(Hash(B.y))`. @kmshin1397 please correct me if I'm wrong?

Reply:
Yes, as far as I know, that's true. So collision should only happen if B.y == DUMMY, and actually not having the DUMMY update would actually lower collision probability since we get rid of that scenario.